src/main/scala/org/apache/spark/sql/connector/kinesis/KinesisV2MicrobatchStream.scala

```diff
@@ -113,10 +113,22 @@ class KinesisV2MicrobatchStream (
       shardInfo.iteratorType,
       shardInfo.iteratorPosition)
 
-    val records = kinesisClient.getKinesisRecords(shardIterator, 1)
-    val recordsSize = records.records.size()
-    val millisBehindLatest = records.millisBehindLatest.longValue()
-    logDebug(s"hasNewData recordsSize ${recordsSize}, millisBehindLatest ${millisBehindLatest}")
+    def getRecordsResponseInfo(shardIterator: String): (Int, Long, String) = {
+      val records = kinesisClient.getKinesisRecords(shardIterator, 1)
+      val recordsSize = records.records.size()
```

Comment:
`records.records` can bull null. Let's check for it.

Reply:
The API `hasRecords()` is misleading. The API never returns `null`. It returns empty array.

Reply:
Ah I see, the service can return null, but the API will not.

> records
>
> public final [List] \<Record\> records()
> The data records retrieved from the shard.
> 
> Attempts to modify the collection returned by this method will result in an UnsupportedOperationException.
> 
> This method will never return null. If you would like to know whether the service returned this field (so that you can differentiate between null and empty), you can use the hasRecords() method.